phc/web/patient_filter_query_builder.py

```diff
@@ -25,38 +25,77 @@ def __getattr__(self, attr):
         return handler
 
     def __global_handler(self, *args, **kwargs):
-        component = {
-            'operator': None,
-            'value': None
-        }
-        for operator, value in kwargs.items():
-            component['operator'] = operator
-            component['value'] = value
-
-        self.component_body[self.__global_handler.__func__.func_name] = [component]
+        if args and len(args) > 0:
+            raise NotImplementedError('Property only support key value arguments!')
+
+        components = []
+        for operator, value_or_values in kwargs.items():
+            values = value_or_values if type(value_or_values) is list else [value_or_values]
+            for value in values:
+                components.append({
+                    'operator': operator,
+                    'value': value
+                })
+
+        self.component_body[self.__global_handler.__func__.func_name] = components
         return self
 
     def to_dict(self):
         return self.component_body
 
 
-class PatientFilterComponentsQueryBuilder(BaseQueryBuilder):
+class ObservationProperty(Property):
+    def __init__(self):
+        Property.__init__(self)
+
+    def with_components(self, components_properties):
+        """
+        Handle component property.
+        """
+        if type(components_properties) is not list:
+            raise TypeError('Observation property with_components parameter must be in an array!')
+
+        self.component_body['components'] = components_properties
+        return self
+
+    def to_dict(self):
+        component_body_dict = dict()
+        for key, value_or_values in self.component_body.items():
+            if key == 'components':
+                if type(value_or_values) is list:
+                    component_values = []
+                    for value in value_or_values:
+                        component_values.append(value.to_dict())
```

Comment:
Minor: this can be rewritten as a comprehension, e.g. `[x.to_dict() for x in value_or_values]`